src/main/Database/OrganizationDaoImpl.java

```diff
@@ -0,0 +1,71 @@
+package Database;
+
+import Config.DeploymentLevel;
+import Config.MongoConfig;
+import Organization.Organization;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import org.bson.types.ObjectId;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static com.mongodb.client.model.Filters.eq;
+
+public class OrganizationDaoImpl implements OrganizationDao {
+  private MongoCollection<Organization> OrganizationCollection;
+
+  public OrganizationDaoImpl(DeploymentLevel deploymentLevel) {
+    MongoDatabase db = MongoConfig.getDatabase(deploymentLevel);
+    if (db == null) {
+      throw new IllegalStateException("DB cannot be null");
+    }
+    OrganizationCollection = db.getCollection("Organization", Organization.class);
+  }
+
+  @Override
+  public Optional<Organization> get(String orgName) {
+    return Optional.ofNullable(OrganizationCollection.find(eq("orgName", orgName)).first());
+  }
+
+  @Override
+  public Optional<Organization> get(ObjectId id) {
+    return Optional.ofNullable(OrganizationCollection.find(eq("_id", id)).first());
+  }
+
+  @Override
+  public List<Organization> getAll() {
+    return OrganizationCollection.find().into(new ArrayList<>());
+  }
+
+  @Override
+  public int size() {
+    return (int) OrganizationCollection.countDocuments();
+  }
+
+  @Override
+  public void delete(Organization Organization) {
+    OrganizationCollection.deleteOne(eq("orgName", Organization.getOrgName()));
```

Comment:
I would move this into a static final variable, like ORG_NAME_FILTER or something